### ISSUE TITLE
Undesired horizontal scrolling in latest Rstudio-server

### ISSUE THREAD

**Issue by kevinrse** [rstudio/rstudio#1813]
I use Rstudio server edition every day. My colleague just updated our internal server, where I do all my work, to the latest version of Rstudio-server (1.1.383). This was two days ago.

Today, on several occasions when editing a script, the editor window scrolls horizontally to keep my cursor at the leftmost edge of the viewable pane! In other words, when I hit "end" on my keyboard (or click, or move the text cursor any other way), the entirety of the line I'm editing is hidden "beyond" the left edge of the text editing pane!

I've never had this happen before, so I'm fairly sure it's a bug. BUT, I've been using Rstudio all day for two days since the version upgrade, and this issue has only happened 2 or 3 times today. (I closed the open instance of Rstudio-server and opened it again in a new browser window; that seemed to fix the problem). I can't replicate the problem "at will;" not sure what leads to the odd behavior.

I hope this report is of some help to you. Thanks for a great product. I run Linux Mint 18.1 Serena Cinnamon Edition and I use Firefox 57.0 to access Rstudio on the server.

**Comment by etienne-s** [comment]
I sometimes face the same issue on RStudio Desktop 1.1.383. Hard to replicate...

**Comment by ronblum** [comment]
Thanks you both for your input. Since this is hard to reproduce, I'll keep an eye on it myself. In the meantime, if you run into this issue again, could you send a screenshot? And any information about what you were doing right before this happens would be helpful.

**Comment by kevinrse** [comment]
OK, just happened again---I've identified the behavior that causes it. Surprised I didn't figure it out the first itme.

I use the keyboard shortcuts ctrl+shift+(1 | 2 | 3) to zoom view to the editor, interactive session and help/plots panes, respectively.

It seems to reliably happen if I FIRST ctrl-shift-3 (zoom to plot or help), THEN ctrl-shift-1 (zoom editor).

The undesired scrolling persists as long as the editor pane is zoomed/maximized. It stops if I repeat ctrl-shift-1 again (restoring the normal three-pane view).

Hope this helps. Thanks again.
 - Kevin

**Comment by ronblum** [comment]
Thank you, @kevinrse. I was able to reproduce the problem. I tested multiple combinations, and it looks like the problem occurs with Linux and Windows browsers (both Firefox and Chrome) pointing to RStudio Server. It works fine with MacOS pointing to a server and with RStudio Desktop on Mac, Windows, and Linux.

_PROBLEM_
OpenSUSE 42.3 (12.3) + Firefox 52.3 —> RStudio Server 1.1.393
Ubuntu 16.04 + Firefox 56.0 —> RStudio Server 1.1.393
Windows 10 + Chrome 62.0 —> RStudio Server 1.1.393

_WORKS FINE_
MacOS 10.12 + Chrome 63.0 —> RStudio Server 1.1.393
MacOS 10.12 + Rstudio Desktop 1.1.393
MacOS 10.12 + RStudio Desktop 1.2.197
Ubuntu 16.04 + RStudio Desktop 1.1.383
Windows 10 + RStudio Desktop 1.1.393

By the way, note that using CTRL+SHIFT+1 again will show a properly working zoomed editor.
